capstone/src/main/java/com/google/step/search/ElasticsearchClient.java

```diff
@@ -78,37 +78,40 @@ public void updateRestaurantHeader(RestaurantHeader restaurantHeader) throws IOE
    */
   @Override
   public List<RestaurantHeader> searchRestaurants(String query) throws IOException {
-    return query.isEmpty() ? getRandomRestaurants() : queryRestaurantHeaders(query);
-  }
-
-  private List<RestaurantHeader> queryRestaurantHeaders(String query) throws IOException {
     List<String> requestPath = Arrays.asList("", RESTAURANTS, "_search");
 
-    String requestBody = new JSONObject()
-        .put("query", new JSONObject()
-            .put("multi_match", new JSONObject()
-                .put("query", query)
-                .put("fields", new JSONArray(Arrays.asList("name", "cuisine")))))
-        .toString();
+    JSONObject queryRequest = query.isEmpty() ? createMatchAllQuery() : createBasicSearchQuery(query);
+    String requestBody = addBoostingToQuery(queryRequest).toString();
```

Comment:
Nice work fixing this logic, this is exactly the kind of thing I was hoping for.